src/drivers/St7789.h

```diff
@@ -2,6 +2,10 @@
 #include <cstddef>
 #include <cstdint>
 
+#include <hal/nrf_gpio.h>
+#include <nrfx_log.h>
+#include "drivers/Spi.h"
+
```

Comment:
Why move thse to the header? Also, if we do move them, we should remove the Spi forward declaration.

Reply:
Honestly not totally sure, I wrote this a while ago. Will move them back I think unless I rediscover why...